src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -32,44 +32,41 @@ public override bool CanHandleType(SqlObjectType objectType)
-                        Processors = prototype.Processors,
-                        IsClustered = prototype.IsClustered,
-                        IsHadrEnabled = prototype.IsHadrEnabled,
-                        IsPolyBaseInstalled = prototype.IsPolyBaseInstalled,
-                        IsXTPSupported = prototype.IsXTPSupported,
-                        Product = prototype.Product,
-                        ReservedStorageSizeMB = prototype.ReservedStorageSizeMB,
-                        RootDirectory = prototype.RootDirectory,
-                        ServerCollation = prototype.ServerCollation,
-                        ServiceTier = prototype.ServiceTier,
-                        StorageSpaceUsageInMB = prototype.StorageSpaceUsageInMB,
-                        Version = prototype.Version,
-                        MinServerMemory = prototype.MinServerMemory,
-                        MaxServerMemory = prototype.MaxServerMemory,
-                        AutoProcessorAffinityMaskForAll = prototype.AutoProcessorAffinityMaskForAll,
-                        AutoProcessorAffinityIOMaskForAll = prototype.AutoProcessorAffinityIOMaskForAll,
-                        NumaNodes = prototype.NumaNodes
-                    };
-                }
-                return Task.FromResult(new InitializeViewResult { ViewInfo = this.serverViewInfo, Context = context });
+                    Name = prototype.Name,
+                    HardwareGeneration = prototype.HardwareGeneration,
+                    Language = prototype.Language,
+                    MemoryInMB = prototype.MemoryInMB,
+                    OperatingSystem = prototype.OperatingSystem,
+                    Platform = prototype.Platform,
+                    Processors = prototype.Processors,
+                    IsClustered = prototype.IsClustered,
+                    IsHadrEnabled = prototype.IsHadrEnabled,
+                    IsPolyBaseInstalled = prototype.IsPolyBaseInstalled,
+                    IsXTPSupported = prototype.IsXTPSupported,
+                    Product = prototype.Product,
+                    ReservedStorageSizeMB = prototype.ReservedStorageSizeMB,
+                    RootDirectory = prototype.RootDirectory,
+                    ServerCollation = prototype.ServerCollation,
+                    ServiceTier = prototype.ServiceTier,
+                    StorageSpaceUsageInMB = prototype.StorageSpaceUsageInMB,
+                    Version = prototype.Version,
+                    MinServerMemory = prototype.MinServerMemory,
+                    MaxServerMemory = prototype.MaxServerMemory,
+                    AutoProcessorAffinityMaskForAll = prototype.AutoProcessorAffinityMaskForAll,
+                    AutoProcessorAffinityIOMaskForAll = prototype.AutoProcessorAffinityIOMaskForAll,
+                    NumaNodes = prototype.NumaNodes
+                };
             }
+            var context = new ServerViewContext(requestParams);
```

Comment:
The connection for this context wasn't being used, and it also already gets disposed when the view itself is disposed.